repositories/org_repository.go

```diff
@@ -34,6 +46,27 @@ func NewOrgRepo(rootNamespace string, privilegedClient client.Client) *OrgRepo {
 	}
 }
 
+func (r *OrgRepo) CreateOrg(ctx context.Context, org OrgRecord) (OrgRecord, error) {
+	anchor := &v1alpha2.SubnamespaceAnchor{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: org.Name,
```

Comment:
we should stay consistent with how we are generating unique id. We have been using google's uuid package for generating them and assigning it to `metadata.name`- see here https://github.com/cloudfoundry/cf-k8s-api/blob/357b9f8d26bcd942e398b22630a8c06c4a4fb087/apis/app_handler.go#L137

Reply:
We discussed our reasoning for this (also written down in [this ADR](https://github.com/cloudfoundry/cf-k8s-api/blob/main/docs/architecture-decisions/0001-cf-resource-guid-format.md)) a bit during the CF on K8s SIG meeting on Tuesday and the general consensus was to use what @akrishna90 is suggesting for all resources since:

1. It aligns with the current GUIDs used by CF so will be more compatible with existing systems
2. Names in CF are mutable (people can change org, space, app names, etc. at any time) and K8s Metadata.Name fields are immutable. This makes it confusing and difficult to handle the situation where a user does rename these resources so the immutable CF GUID feels like a natural fit.

Anyways, wanted to share a bit more context from that SIG meeting about why we want this change since not everyone involved with this PR was in attendance.